SpellCheck/src/main/java/AIR/SpellCheck/Engines/HunspellEngine.java

```diff
@@ -25,44 +27,52 @@
 @Component
 public class HunspellEngine implements ISpellEngine
 {
-  
+  private final static Logger LOG = LoggerFactory.getLogger(HunspellEngine.class);
+
+  private final SpellEngineFactory spellEngineFactory;
+
   @Autowired
-  SpellEngineFactory _spellEngine;
-  
+  public HunspellEngine(final SpellEngineFactory spellEngineFactory) {
+    this.spellEngineFactory = spellEngineFactory;
+  }
+
   @Override
-  public boolean CheckWord (String word) {
+  public boolean CheckWord (final String word) {
+    final String languageCode = getLanguage();
     boolean isCorrect = true;
     try {
-      String languageCode = WebHelper.getQueryString ("lang");
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
+      if (spellEngineFactory.getDictionary (languageCode).misspelled(word)) {
         isCorrect = false;
       }
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
-        isCorrect = false;
-      }
-    } catch (Exception e) {
-      e.printStackTrace();
+    } catch (final Exception e) {
+      LOG.error("Unable to spell check word {} in language {}", word, languageCode);
     }
     return isCorrect;
   }
   
   
   @Override
-  public List<String> GetSuggestions (String word)  {
-    List<String> suggestions = new ArrayList<String> ();
+  public List<String> GetSuggestions(final String word)  {
+    final String languageCode = getLanguage();
+    final List<String> suggestions = new ArrayList<> ();
     try {
-      String languageCode = WebHelper.getQueryString ("lang");
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
-        suggestions.addAll (_spellEngine.getDictionary (languageCode).suggest(word));
-      }
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
-        suggestions.addAll (_spellEngine.getDictionary (languageCode).suggest(word));
+      if (spellEngineFactory.getDictionary (languageCode).misspelled(word)) {
+        suggestions.addAll (spellEngineFactory.getDictionary (languageCode).suggest(word));
       }
```

Comment:
I know this is a slight change in functionality.  Previously, the code was returning all the suggestions twice... but I just couldn't make myself do that... Any objections?